draft-ietf-teep-architecture.md

```diff
@@ -687,17 +690,22 @@ and TEE for TAM trust verification and TA signature verification.
 1. Build two apps:          |       |                |            |
                             |       |                |            |
    (a) Untrusted            |       |                |            |
-       App - 2a. Supply --> | --- 3. Install ------> |            |
+       App - 2a. Supply --->|<---------- 3. Get App -|            |
+                            |----------------------->|            |
```

Comment:
--> means the source or destination? We use it as destination: A-->B means that the object will be installed in B. To this, I think we should also change "3. Get App to use -->", and 4 below "-- 4. Get TA -->".

Reply:
The arrow directions confused Daniel who expected them to show who initiated the step.  Since Get App is initiated from the device not the TAM, the old 3 Install was confusing in his review.  Hence the change to show it initiates from the device in this example, and the TAM returns the app in 694.  That was my attempt to address his comment.

Reply:
If it means who initiates, "Get an app cert" should use the other direction: an app developer goes to CA to get a cert where developer initiates the certificate enrollment and requests. Similarly, "Get a TAM cert" should be that TAM initiates enrollment to a CA. I am fine either convention but feel that the "Get App" and "Get Cert" are using different conventions. The original "3. Install" is consistent with "2. supply". Can we clarify with Daniel on this without change? Or we change the arrows for the "cert" arrows?

Reply:
Good point, I have changed the arrows back and instead added a parenthetical phrase to clarify.  How's it look now?

Reply:
I like the explanation in the text. Sorry for one more comment. Do we have to change "install app" to "get app"? "Get app to" is what this means. In the protocol spec, we do use API "TrustedAppInstall"; we don't use "get app". TAM sends app to TEE, not a TEE to go out to get an app. The original word "install app" sounds right, in particular, in coherence with the arrow. Your parenthetical phrase resolve Daniel's comment. Can this work? I am OK if you strongly like to use "Get App".